x/mongo/driver/batch_cursor.go

```diff
@@ -241,7 +248,7 @@ func (bc *BatchCursor) getMore(ctx context.Context) {
 
 	// Required for legacy operations which don't support limit.
 	numToReturn := bc.batchSize
-	if bc.limit != 0 && bc.numReturned+bc.batchSize > bc.limit {
```

Comment:
Thanks for catching this bug @liuliqiang ! We agree that this if check is not correct, and that if `bc.numReturned == bc.limit` and `bc.batchSize == 0` (as it does by default), the if check will not be entered, and another `getMore` will be run when the cursor should have been closed.

However, it looks like the fix could simply be changing this line to
```
if bc.limit != 0 && bc.numReturned+bc.batchSize >= bc.limit {
```
Could you modify the if check as above instead of using the `minInt32()` logic?

Reply:
well, it seems work.
And we found that you have a new pr with uts.
I will disable this pr now.